src/ipAllowed.js

```diff
@@ -11,22 +11,45 @@ module.exports = function(allowedSubnets) {
   assert(allowedSubnets, 'Must provide allowed subnets in CIDR format');
   assert(Array.isArray(allowedSubnets), 'Allowed subnets must be a list');
 
-  let subnets = [];
+  let subnetsV4 = [];
+  let subnetsV6 = [];
 
   for (let subnet of allowedSubnets) {
-    subnets.push(ipaddr.parseCIDR(subnet));
+    let parsed = ipaddr.parseCIDR(subnet);
+    // Let's be sure that this value is a 2-tuple of an object and a mask
+    // length.  Probably overkill
+    assert(typeof parsed[0] === 'object');
+    assert(typeof parsed[1] === 'number');
+    assert(parsed.length === 2);
+    if (parsed[0].kind() === 'ipv4') {
+      subnetsV4.push(parsed);
+    } else if (parsed[0].kind() === 'ipv6') {
+      subnetsV6.push(parsed);
+    } else {
+      assert(false, 'CIDR Specifies unknown address type');
```

Comment:
I did `assert(false)` only to match the other initialization assertions being thrown using assert.